internal/app/service.go

```diff
@@ -737,45 +740,44 @@ func (svc *Service) BuildContext(correlationId string, contentType string) inter
 	return appfunction.NewContext(correlationId, svc.dic, contentType)
 }
 
-// Publish pushes data to the messagebus
-func (svc *Service) Publish(v any) error {
+// Publish pushes data to the Messagebus using configured topic
+func (svc *Service) Publish(data any) error {
 	messageClient := bootstrapContainer.MessagingClientFrom(svc.dic.Get)
 	if messageClient == nil {
-		return fmt.Errorf("messagebus is disabled via configuration")
+		return fmt.Errorf(messageBusDisabledErr)
 	}
 
 	triggertopic := svc.config.Trigger.PublishTopic
 	baseTopic := svc.config.MessageBus.BaseTopicPrefix
 
-	payload, err := json.Marshal(v)
+	payload, err := json.Marshal(data)
 	if err != nil {
-		return fmt.Errorf("failed to marshal data: %v", err)
+		return fmt.Errorf("failed to marshal data for publishing: %v", err)
 	}
 	message := types.NewMessageEnvelope(payload, context.Background())
 	err = messageClient.Publish(message, contractsCommon.BuildTopic(baseTopic, triggertopic))
 	if err != nil {
-		return fmt.Errorf("failed to publish data to messagebus: %v", err)
+		return fmt.Errorf("%v: %v", publishDataErr, err)
 	}
 	return nil
 }
 
-// Publish pushes data to the messagebus for a given topic
+// Publish pushes data to the MessageBus using given topic
 func (svc *Service) PublishWithTopic(topic string, data any) error {
 	messageClient := bootstrapContainer.MessagingClientFrom(svc.dic.Get)
 	if messageClient == nil {
-		return fmt.Errorf("messagebus is disabled via configuration")
+		return fmt.Errorf(messageBusDisabledErr)
 	}
 
-	full_topic := commonConstants.BuildTopic(svc.config.MessageBus.BaseTopicPrefix, topic)
 	payload, err := json.Marshal(data)
 	if err != nil {
-		return fmt.Errorf("failed to marshal data: %v", err)
+		return fmt.Errorf("failed to marshal data for publishing: %v", err)
 	}
 
 	message := types.NewMessageEnvelope(payload, context.Background())
-	err = messageClient.Publish(message, full_topic)
+	err = messageClient.Publish(message, commonConstants.BuildTopic(svc.config.MessageBus.BaseTopicPrefix, topic))
```

Comment:
Since this isn't referencing a constant... ;-)

```suggestion
	err = messageClient.Publish(message, coreCommon.BuildTopic(svc.config.MessageBus.BaseTopicPrefix, topic))
```